### PR TITLE
Add livenessProbe and startupProbe to worker pool containers

### PR DESCRIPTION
**Pull request by jcardonnet**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

**Comment by madhavajay** [comment]
@yashgorana can you review this?

**Comment by yashgorana** [comment]
@madhavajay do we want these values to be configured? if not we can merge it.

File: packages/syft/src/syft/custom_worker/runner_k8s.py [hunk 1]
```diff
@@ -230,6 +230,24 @@ def _create_stateful_set(
                                     "image": tag,
                                     "env": env_vars,
                                     "volumeMounts": volume_mounts,
+                                    "livenessProbe": {
+                                        "httpGet": {
+                                            "path": "/api/v2/?probe=livenessProbe",
+                                            "port": 80,
+                                        },
+                                        "initialDelaySeconds": 30,
```
Comment (yashgorana) [inline packages/syft/src/syft/custom_worker/runner_k8s.py]: don't need "initialDelaySeconds" if we're using startupProbe.

Comment (jcardonnet) [reply packages/syft/src/syft/custom_worker/runner_k8s.py]: Ok. I'll remove it

**Comment by jcardonnet** [comment]
Tested `livenessProbe` by creating a custom image that deliberately raises an exception on startup.
container_fail

This are the relevant parts of  `kubectl describe pod <podname>` 

```
Name:             backend-devspace-ccc49dd7-7crpp
Namespace:        syft
Priority:         0
Service Account:  backend-service-account
Node:             k3d-syft-dev-server-0/172.25.0.3
Start Time:       Thu, 20 Jun 2024 15:21:44 -0300

Annotations:      <none>
Status:           Running
IP:               10.42.0.15
IPs:
  IP:           10.42.0.15
Controlled By:  ReplicaSet/backend-devspace-ccc49dd7
Containers:
  backend-container:
    Container ID:   containerd://b11b3720c23b0d1f1c65e7b05288db3cdfc16fb2c5012247dc64a9166cac98f6
    Image:          k3d-registry.localhost:5800/openmined/grid-backend:dev-1718907620
    Image ID:       k3d-registry.localhost:5800/openmined/grid-backend@sha256:d42bad761a8b0c476be03be5cbea7eb382a19957e0bc9eb9d8a33723acb6c2fb
    Port:           80/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Thu, 20 Jun 2024 15:37:25 -0300
    Last State:     Terminated
      Reason:       Completed
      Exit Code:    0
      Started:      Thu, 20 Jun 2024 15:32:25 -0300
      Finished:     Thu, 20 Jun 2024 15:37:25 -0300
    Ready:          False
    Restart Count:  3
    Liveness:       http-get http://:api-port/api/v2/%3Fprobe=livenessProbe delay=0s timeout=5s period=15s #success=1 #failure=3
    Startup:        http-get http://:api-port/api/v2/metadata%3Fprobe=startupProbe delay=0s timeout=1s period=10s #success=1 #failure=30

Conditions:
  Type              Status
  Initialized       True
  Ready             False
  ContainersReady   False
  PodScheduled      True

Events:
  Type     Reason     Age                 From               Message
  ----     ------     ----                ----               -------
  Normal   Scheduled  16m                 default-scheduler  Successfully assigned syft/backend-devspace-ccc49dd7-7crpp to k3d-syft-dev-server-0
  Normal   Pulling    16m                 kubelet            Pulling image "k3d-registry.localhost:5800/openmined/grid-backend:dev-1718907620"
  Normal   Pulled     16m                 kubelet            Successfully pulled image "k3d-registry.localhost:5800/openmined/grid-backend:dev-1718907620" in 40.17s (40.17s including waiting)
  Normal   Created    16m                 kubelet            Created container backend-container
  Normal   Started    16m                 kubelet            Started container backend-container
  Warning  Unhealthy  97s (x87 over 15m)  kubelet            Startup probe failed: Get "http://10.42.0.15:80/api/v2/metadata?probe=startupProbe": dial tcp 10.42.0.15:80: connect: connection refused
```

File: packages/syft/src/syft/custom_worker/runner_k8s.py (hunk 1 above)
Comment (yashgorana) [reply packages/syft/src/syft/custom_worker/runner_k8s.py]: It's still here :D - Meanwhile i'm approving the PR - please merge it once updated

**Review by yashgorana — APPROVED** [review]

File: packages/syft/src/syft/custom_worker/runner_k8s.py (hunk 1 above)
Comment (jcardonnet) [reply packages/syft/src/syft/custom_worker/runner_k8s.py]: Ouch! I didn't notice my `push` had failed. Fixed. Sorry for the fumble

**Pull request merged**